src/main/java/edu/harvard/iq/dataverse/util/FileUtil.java

```diff
@@ -501,24 +501,30 @@ public static String determineFileType(File f, String fileName) throws IOExcepti
 
         if ("application/x-gzip".equals(fileType)) {
             logger.fine("we'll run additional checks on this gzipped file.");
-            // We want to be able to support gzipped FITS files, same way as
-            // if they were just regular FITS files:
-            FileInputStream gzippedIn = new FileInputStream(f);
-            // (new FileInputStream() can throw a "filen not found" exception;
-            // however, if we've made it this far, it really means that the 
-            // file does exist and can be opened)
-            InputStream uncompressedIn = null; 
-            try {
-                uncompressedIn = new GZIPInputStream(gzippedIn);
-                if (isFITSFile(uncompressedIn)) {
-                    fileType = "application/fits-gzipped";
-                }
-            } catch (IOException ioex) {
-                if (uncompressedIn != null) {
-                    try {uncompressedIn.close();} catch (IOException e) {}
+            try (FileInputStream gzippedIn = new FileInputStream(f)) {
+                // (new FileInputStream() can throw a "file not found" exception;
+                // however, if we've made it this far, it really means that the
+                // file does exist and can be opened)
+                InputStream uncompressedIn = null;
+                try {
+                    uncompressedIn = new GZIPInputStream(gzippedIn);
```

Comment:
It wasn't in the issue but would you be willing to do the same thing to this stream? I think you should be able to put this in line 504 in the same try.

Reply:
I've updated the PR